backend/capellacollab/projects/capellamodels/modelsources/git/routes.py

```diff
@@ -70,22 +69,44 @@ def verify_valid_path_sequences(path: str):
     )
 
 
-@router.post("/", response_model=ResponseGitSource)
+def verify_valid_git_model_on_project_and_capella_model(
+    db: Session, project_slug: str, model_slug: str, git_model_id
+):
+    capella_model = capella_model_crud.get_model_by_slug(
+        db, project_slug, model_slug
+    )
+
+    for git_model in capella_model.git_models:
+        if git_model.id == git_model_id:
+            return
+
+    return HTTPException(
+        status_code=400,
+        detail={
+            "err_code": "git_model_not_exists_on_project_and_model",
+            "reason": "The git model does not exists on the capella model for the project",
+        },
+    )
```

Comment:
I would have used an injectable function that loads a model (https://fastapi.tiangolo.com/tutorial/dependencies/). For example:

```python
def load_capella_model(
  db: Session), project_slug: str, model_slug: str
) -> DatabaseCapellaModel:
  # the code, with a HTTPException if needed
  return capella_model


def load_git_model(
  db: Session, capella_model: DatabaseCapellaModel, git_model_id: int
) -> DatabaseGitModel:
  # the code, with a HTTPException if needed
  return git_model


@router.get()
def route_handling(
  db: Session = Depends(get_db),
  capella_model: DatabaseCapellaModel = Depends(load_capella_model),
  git_model: DatabaseGitModel = Depends(load_git_model)
) :
  # the code
```
I’m not sure of the dependency between the different functions but that’s the idea.